### ISSUE TITLE
Restyle Frontend - Events Page Widget

### ISSUE THREAD

**Issue by minhngo3818** [Computer-Science-Club-OCC/Computer-Science-Club-OCC.github.io#99]
The old ui for events page uses purely css. However, we are going to use `Tailwind css` as our main styling framework. In this issue, event widget need replacing old css classes with tailwind css, and style it with the new design . Orange is `#ff9617`. Hover effect and animation are up to programmers.

This task can be assigned by many contributors in order to select the best design.

Any ideas for better design are welcomed. Here is the design to rely on
<img src="https://user-images.githubusercontent.com/78233689/195009994-650e02b3-fef3-4991-931c-92771aa1fa7f.png" />

**Comment by mfonPeeter** [comment]
Please, can this be assigned to me? 

Also, I think the new design above is okay for the event page considering the fact that it aligns with the primary color of the website.

**Comment by mfonPeeter** [comment]
Thanks for assigning this to me. I will get to work as soon as possible.

But please, can you explain further when you say I should create my branch from ```minhngo-events-optimize-cleanup```, then name it as ```<github_name or issue_number>-restyle-event-widget``` before I start.

Like how do I create my own branch from  ```minhngo-events-optimize-cleanup```. I'm used to creating from the main branch and not from other branches. Please help, so I can start implementing the designs.

Also, just to be clear, are you saying I should name my branch by the github name or the issue number of this issue?

**Comment by minhngo3818** [comment]
Sorry for stating clear branch name. I will make announcement for naming branch. Your branch will be `<your_name>-restyle-event-widget`. 

To create a branch from a different branch, in terminal use `git checkout -b <your-branch> minhngo-events-optimize-cleanup`. Or, if you use vscode, go to git icon ->click three dot -> create a branch from -> name your branch -> select the branch to want to create from. Make sure you fetch the repo first.

**Comment by mfonPeeter** [comment]
Also, am I allowed to remove some styles that don't have any effect on the website? Some styles, like ```overflow: auto``` for example don't have any effect in some places. Or should I implement them with Tailwind CSS? 

I'm asking to know if you implemented some of these styles for future purposes or if there were for testing so I will know if to get rid of them or not.

**Comment by mfonPeeter** [comment]
I'm considering restyling the Events Header from this: 

![before](https://user-images.githubusercontent.com/105684251/195592685-2285790e-0020-4c7a-a87f-a6bb36141e15.png)

to this: 

![after](https://user-images.githubusercontent.com/105684251/195592743-15471da7-03cd-44a9-b358-7b3e24626241.png)

The second looks more appealing to me. If you still want to go with the original one, it is still okay with me.

**Comment by KHVBui** [comment]
> @minhngo3818, I noticed that the website has horizontal scrolling which looks unappealing. After some debugging, I noticed that it is coming from the styles in the body element. Currently, these are the styles in the body element:
> 
> ```
> html,
> body {
> 	height: 100vh;
> 	width: 100vw;
> 	margin: 0;
> 	padding: 0;
> 	background-color: #21262d;
> 	background-size: cover;
> }
> ```
> 
> I noticed changing the width from 100vw to 100% fixes the issue. Please can I implement this even though it is not part of the issue I'm focused on? I'm asking because any design I have to make has to put the horizontal scrolling into consideration and removing it will be very convenient and also make the website more appealing.
> 
> But if I'm to implement this, it will be with custom CSS since I'm not changing the body styles from custom CSS to Tailwind CSS

To be clear, you are removing horizontal scrolling? That is okay with me. Just be sure it doesn't create breaking changes on other pages.

**Comment by KHVBui** [comment]
> I'm considering restyling the Events Header from this:
> 
> ![before](https://user-images.githubusercontent.com/105684251/195592685-2285790e-0020-4c7a-a87f-a6bb36141e15.png)
> 
> to this:
> 
> ![after](https://user-images.githubusercontent.com/105684251/195592743-15471da7-03cd-44a9-b358-7b3e24626241.png)
> 
> The second looks more appealing to me. If you still want to go with the original one, it is still okay with me.

Let's keep the original header for now. Restyling that can be a different issue, and I'd rather we wait for the UI/UX team's input before we make changes.

**Comment by minhngo3818** [comment]
@mfonPeeter Yes, you can change body element. Since this is a restyling task, you can freely to modify codes that may relate to your issue. 

I keep my old style to give you a reference. Because we use tailwind from now, so those styles are deprecated. Once your design is approved, old css files related to the component will be removed. Unless, you use those css files to compact tailwind css class into a class name (function directives), then you just need to replace my old styles with tailwind and keep the file.

The header is ok to change.

**Comment by mfonPeeter** [comment]
Thanks a lot. I will adhere to everything you have said.

**Comment by mfonPeeter** [comment]
And yes the horizontal scrolling was addressed in PR #104. But I will just apply it to my own since it seems it has not yet been merged

**Comment by KHVBui** [comment]
> And yes the horizontal scrolling was addressed in PR #104. But I will just apply it to my own since it seems it has not yet been merged

I'm in the process of merging it right now, so I'd recommend using that styling application.

**Comment by mfonPeeter** [comment]
Okay @KHVBui

**Comment by mfonPeeter** [comment]
I currently have an issue. I first downloaded the zip file of the repo and started working on it just to test things out. I'm done with restyling the frontend of the events widget using tailwind CSS framework. Now I want to create my own branch from ```minhngo-events-optimize-cleanup```.

I first copied the link and cloned the repo, then I did a ```git branch -r``` to see the branches in the remote repo. I then did a ```git checkout -b mfonPeeter-restyle-event-widget origin/minhngo-events-optimize-cleanup```. It successfully created a new branch in the the ```minhngo-events-optimize-cleanup``` branch. 

I then tried copying and pasting the styles I've already done, but it had no effect. The styles were not showing. I'm really lost here, because I'm wondering why it is not having an effect. 

I switched back to the main branch and tried applying the styles, but still to no effect. 

Just a side note, I didn't fork the repo, I just copied the link of this repo and cloned it. I did this because that was the only way I could get access to the remote branches. 

I've been on this for three days now with no significant progress. Please, I need some help. Thanks.

**Comment by KHVBui** [comment]
@mfonPeeter How are you running the client right now? Are you using `npm start` or doing the docker commands?
If doing the docker commands, temporary local changes won't be reflected, so I recommend just doing the basic `npm start` for the client.

**Comment by minhngo3818** [comment]
@mfonPeeter  Maybe creating a new branch will not add dependencies that you have been working with. It happens sometime. Try to run `npm install` in the new branch and then start client again. Hope it solves your problem.

**Comment by mfonPeeter** [comment]
> @mfonPeeter How are you running the client right now? Are you using `npm start` or doing the docker commands? If doing the docker commands, temporary local changes won't be reflected, so I recommend just doing the basic `npm start` for the client.

Oh no, I'm not making use of docker commands. I'm using ```npm start````

**Comment by mfonPeeter** [comment]
> @mfonPeeter Maybe creating a new branch will not add dependencies that you have been working with. It happens sometime. Try to run `npm install` in the new branch and then start client again. Hope it solves your problem.

I will try this out and get back to you. Thanks

**Comment by mfonPeeter** [comment]
> @mfonPeeter Maybe creating a new branch will not add dependencies that you have been working with. It happens sometime. Try to run `npm install` in the new branch and then start client again. Hope it solves your problem.

So I tried running `npm install` again, but it wasn't working. I then cloned the repo again and couldn't find the different components that were there before such as `EventBrief.jsx` and the like. When I then looked at the `Events.jsx` component, they were still there as imports, but I can't find them again. As a result, I can't even change the style since I can't find the components. Please, what should I do here?

I was also thinking if I should just use the normal fork, clone and then create a new branch method. Then when creating a pull request, I will create it in the `minhngo-events-optimize-cleanup` branch. 

Below is a picture just to show that I actually finished with it:

![Screenshot (25)](https://user-images.githubusercontent.com/105684251/198955147-9411073d-3054-44f5-b70a-11bebec2fa81.png)

![Screenshot (26)](https://user-images.githubusercontent.com/105684251/198955160-6f9900a6-425b-488c-83fa-d9beb365a2ab.png)

**Comment by KHVBui** [comment]
@mfonPeeter 
PR #107 moved all of the event components into one file:
client/src/components/events/events.jsx

Styling has also been moved to
client/src/pages/events/events-page.css

Also, what code editor are you using? I use VS Code together with GitHub Desktop for my workflow.

**Comment by mfonPeeter** [comment]
@KHVBui 
The different event components were used in client/src/components/events/events.jsx. If we look at how they were being used, we notice that props were being passed meaning that they were created somewhere else and not in client/src/components/events/events.jsx.  So, I can't find where they were created in the files.

Also, when I open client/src/pages/events/events-page.css in my code editor, it is only two classes I see that are being designed. And I can't find the styles for the rest.

I'm currently making use of VS Code as my editor.

**Comment by KHVBui** [comment]
@mfonPeeter 
I always try to create a feature branch from `main` and then in that branch do a PR to `main`.

Apologies, the styling for the components is contained in `client/src/components/events/events.css`

Event components are currently created in `client/src/components/events/events.jsx` though.
We know this because they are exported from that file. 
<img width="244" alt="Screen Shot 2022-11-03 at 12 08 57 AM" src="https://user-images.githubusercontent.com/36748732/199664346-112447ba-81ee-4643-bc87-52c7c513e757.png">

**Comment by minhngo3818** [comment]
@mfonPeeter @KHVBui, sorry for all trouble. 

The components are stored in `client/src/components/events/events.jsx`. I packed all event components in one file for optimization. 
The styles for all event components are in `client/src/components/events/events.css`
The file `client/src/page/events/events-page.css` stores styles for the events, not components.

**Comment by mfonPeeter** [comment]
Sorry, for the late reply. But my discord username is mfonPeeter#6053. I will be expecting your message. Thanks @KHVBui

**Comment by minhngo3818** [comment]
This task is postponed. Please wait for the final design. @mfonPeeter There won't be any major differences from the design you have been working on.

**Comment by minhngo3818** [comment]
This issue will be addressed in a new issue with a new design for the event widget.
